pkg/manager/manager.go

```diff
@@ -134,21 +149,51 @@ func (m *SupportBundleManager) Run() error {
 		},
 	}
 
-	for i, phase := range phases {
-		logrus.Infof("Running phase %s", phase.Name)
-		m.status.SetPhase(phase.Name)
-		if err := phase.Run(); err != nil {
-			m.status.SetError(err.Error())
-			logrus.Errorf("Failed to run phase %s: %s", phase.Name, err.Error())
-			break
+	m.runAllPhases(requiredPhases, optionalPhases, postPhases)
+
+	<-m.context.Done()
+	return nil
+}
+
+func (m *SupportBundleManager) runAllPhases(requiredPhases []RunPhase, optionalPhases []RunPhase, postPhases []RunPhase) {
+	progressCount := 0
+	maxProgressCount := len(requiredPhases) + len(optionalPhases) + len(postPhases)
+
+	for _, phase := range requiredPhases {
+		if err := m.runPhase(phase, &progressCount, maxProgressCount); err != nil {
+			logrus.Errorf("Failed to run requiredPhases %s: %s", phase.Name, err.Error())
+			return
 		}
+	}
 
-		progress := 100 * (i + 1) / len(phases)
-		m.status.SetProgress(progress)
-		logrus.Infof("Succeed to run phase %s. Progress (%d).", phase.Name, progress)
+	for _, phase := range optionalPhases {
+		if err := m.runPhase(phase, &progressCount, maxProgressCount); err != nil {
+			logrus.Errorf("Failed to run optionalPhases %s: %s", phase.Name, err.Error())
+			// Since it's optional, don't return error.
```

Comment:
I believe this change is related to the last commit. Can we fixup this commit to keep the history clean?

Reply:
Fixed.